providers/multipart/src/main/java/org/jboss/resteasy/plugins/providers/multipart/Mime4JWorkaround.java

```diff
@@ -0,0 +1,77 @@
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.jboss.resteasy.plugins.providers.multipart;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.james.mime4j.MimeException;
+import org.apache.james.mime4j.MimeIOException;
+import org.apache.james.mime4j.codec.DecodeMonitor;
+import org.apache.james.mime4j.dom.Message;
+import org.apache.james.mime4j.field.DefaultFieldParser;
+import org.apache.james.mime4j.field.LenientFieldParser;
+import org.apache.james.mime4j.message.BasicBodyFactory;
+import org.apache.james.mime4j.message.BodyFactory;
+import org.apache.james.mime4j.message.DefaultBodyDescriptorBuilder;
+import org.apache.james.mime4j.message.MessageImpl;
+import org.apache.james.mime4j.parser.MimeStreamParser;
+import org.apache.james.mime4j.stream.BodyDescriptorBuilder;
+import org.apache.james.mime4j.stream.MimeConfig;
+
+/**
+ * Copy code from org.apache.james.mime4j.message.DefaultMessageBuilder.parseMessage().
+ * Alter said code to use Mime4JWorkaroundBinaryEntityBuilder instead of EntityBuilder.
+ */
+public class Mime4JWorkaround {
+    /**
+     * This is a rough copy of DefaultMessageBuilder.parseMessage() modified to use a Mime4JWorkaround as the contentHandler instead
+     * of an EntityBuilder.
+     * <p>
+     *
+     * @param is
+     * @return
+     * @throws IOException
+     * @throws MimeIOException
+     * @see org.apache.james.mime4j.message.DefaultMessageBuilder#parseMessage(java.io.InputStream)
+     */
+    public static Message parseMessage(InputStream is) throws IOException, MimeIOException {
+        try {
+            MessageImpl message = new MessageImpl();
+            MimeConfig cfg = new MimeConfig();
+            boolean strict = cfg.isStrictParsing();
+            DecodeMonitor mon = strict ? DecodeMonitor.STRICT : DecodeMonitor.SILENT;
+            BodyDescriptorBuilder bdb = new DefaultBodyDescriptorBuilder(null, strict ? DefaultFieldParser.getParser() : LenientFieldParser.getParser(), mon);
+            BodyFactory bf = new BasicBodyFactory();
```

Comment:
I have a feeling this could be too much of a simplification; we used to have a feature for writing big messages to the disk instead of using the memory, which I believe this PR would remove if merged as is.

Reply:
See org.apache.james.mime4j.storage.StorageBodyFactory for example